### PR TITLE
Bump deps and fix Aruba 0.9 deprecations

### PR DESCRIPTION
**Pull request by chastell**
(That thing where you want to get back to hacking on reek, think ‘I’ll just bring deps to the current upstreams’ and spend an hour fixing Aruba deprecations.)

### CODE REVIEW

File: features/step_definitions/reek_steps.rb [hunk 1]
```diff
@@ -11,69 +11,72 @@
 end
 
 Then /^it reports nothing$/ do
-  assert_exact_output '', all_stdout
+  expect(last_command_started).to have_output_on_stdout('')
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: I'm trying to wrap my head around the code in Aruba starting at https://github.com/cucumber/aruba/blob/bfb67969567cf9d74140c1957e58cf65251adda7/lib/aruba/cucumber/command.rb#L134, but I *think* we need something like `an_output_string_being_eq` in here.

Comment (mvz) [inline features/step_definitions/reek_steps.rb]: Or, maybe we should do what this line does: https://github.com/cucumber/aruba/blob/bfb67969567cf9d74140c1957e58cf65251adda7/lib/aruba/cucumber/command.rb#L312

Comment (chastell) [inline features/step_definitions/reek_steps.rb]: Hm, the above switch was what [the deprecation message said we should use](https://github.com/cucumber/aruba/blob/v0.9.0/lib/aruba/api/deprecated.rb#L560) and seems to work (and read) right to me…

Comment (mvz) [inline features/step_definitions/reek_steps.rb]: Yes, it is probably right. In Aruba's case, 'use the source' seems like a bad advice at the moment :smile:.

File: features/rake_task/rake_task.feature [hunk 2]
```diff
@@ -36,6 +39,9 @@ Feature: Reek can be driven through its Task
         [4, 5]:Smelly#m calls @foo.bar 2 times (DuplicateMethodCall)
         [4, 5]:Smelly#m calls puts(@foo.bar) 2 times (DuplicateMethodCall)
         [3]:Smelly#m has the name 'm' (UncommunicativeMethodName)
+
+
+!!! `reek` has found smells - exiting!
```
Comment (mvz) [inline features/rake_task/rake_task.feature]: What does this do?

Comment (chastell) [inline features/rake_task/rake_task.feature]: We [do output this on smell finds](https://github.com/troessner/reek/blob/bump_deps/lib/reek/rake/task.rb#L108) and Cucumber examples can’t indent expected output properly if there are empty lines (regardless whether they’re indented or not).

We can try either switching the `it reports` step to not be an exact match or stop displaying this line on failures; I don’t have a better solution (given that we can’t make the scenario look nice).

File: features/step_definitions/reek_steps.rb [hunk 3]
```diff
@@ -11,69 +11,72 @@
...
 end
 
 Then /^there is no output on stdout$/ do
-  assert_exact_output '', all_stdout
+  expect(last_command_started).to have_output_on_stdout('')
 end
 
 Then /^stdout includes "(.*)"$/ do |text|
-  assert_partial_output(text, all_stdout)
+  expect(last_command_started).to have_output(/#{Regexp.escape(text)}/)
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: You should use `have_output_on_stdout` here.

File: features/step_definitions/reek_steps.rb [hunk 4]
```diff
@@ -11,69 +11,72 @@
...
 
 Then /^the exit status indicates smells$/ do
-  assert_exit_status Reek::CLI::Application::STATUS_SMELLS
+  smells = Reek::CLI::Application::STATUS_SMELLS
+  expect(last_command_started).to have_exit_status(smells)
 end
 
 Then /^it reports:$/ do |report|
-  assert_exact_output "#{report}\n", all_stdout
+  expect(last_command_started).to have_output(report.gsub('\n', "\n"))
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: You should use `have_output_on_stdout` here. That way, you don't have to add the extra line `!!! reek has found smells - exiting!` to the scenario's.

File: features/step_definitions/reek_steps.rb [hunk 5]
```diff
@@ -11,69 +11,72 @@
...
 end
 
 Then /^it reports an error$/ do
-  expect(all_stderr).to_not be_empty
+  expect(all_commands.map(&:stderr).join("\n")).to_not be_empty
 end
 
 Then /^it reports the error ['"](.*)['"]$/ do |string|
-  assert_partial_output string, all_stderr
+  expect(last_command_started).to have_output(/#{Regexp.escape(string)}/)
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: -> `on_stderr`.

File: features/step_definitions/reek_steps.rb [hunk 6]
```diff
@@ -11,69 +11,72 @@
...
 end
 
 Then /^it reports no errors$/ do
-  assert_exact_output '', all_stderr
+  expect(last_command_started).to have_output_on_stderr('')
 end
 
 Then /^it reports an error$/ do
-  expect(all_stderr).to_not be_empty
+  expect(all_commands.map(&:stderr).join("\n")).to_not be_empty
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: Mabe use `last_command_started.stderr`?

**Comment by chastell** [comment]
Aaaah, thanks, @mvz, for the right pointers – all fixed and green.

File: features/step_definitions/reek_steps.rb [hunk 7]
```diff
@@ -11,69 +11,72 @@
...
 
 Then /^it reports:$/ do |report|
-  assert_exact_output "#{report}\n", all_stdout
+  expect(last_command_started).to have_output_on_stdout(report.gsub('\n', "\n"))
 end
 
 Then /^it reports this yaml:$/ do |expected_yaml|
   expected_warnings = YAML.load(expected_yaml.chomp)
-  actual_warnings = YAML.load(all_stdout)
+  actual_warnings = YAML.load(all_commands.map(&:stdout).join("\n"))
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: Sorry to be so nitpicky, but ...
Why not just use `last_comand_started.stdout` here as well?

File: features/step_definitions/reek_steps.rb [hunk 8]
```diff
@@ -11,69 +11,72 @@
...
   expect(actual_warnings).to eq expected_warnings
 end
 
 Then /^it reports this JSON:$/ do |expected_json|
   expected_warnings = JSON.parse(expected_json.chomp)
-  actual_warnings = JSON.parse(all_stdout)
+  actual_warnings = JSON.parse(all_commands.map(&:stdout).join("\n"))
```
Comment (mvz) [inline features/step_definitions/reek_steps.rb]: ... and here?

**Comment by chastell** [comment]
@mvz nitpicky is always good – and the only reason was that the deprecation warning was ‘stop using `all_stdout`, use `all_commands.map(&:stdout).join("\n")` instead’. Fixed, as `last_comand_started.stdout` is clearly better. :)

**Comment by mvz** [comment]
Excellent! :shipit:

**Pull request merged**